### PR TITLE
Ret 2542

### PR DESCRIPTION
**Pull request by pats-john**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,7 +22,7 @@
     "fortifyScan": "./src/test/java/gradlew -p src/test/java fortifyScan",
     "test:pact:run-and-publish": "yarn test:pact && yarn publish-pact",
     "sonar-scan": "sonar-scanner",
-    "test:functional": "NODE_TLS_REJECT_UNAUTHORIZED=0 codeceptjs run --steps -c src/test/end-to-end/codecept.conf.js --steps --reporter mocha-multi --verbose --grep @RET-BAT",
+    "test:functional": "NODE_TLS_REJECT_UNAUTHORIZED=0 codeceptjs run --steps -c src/test/end-to-end/codecept.conf.js --steps --reporter mocha-multi --verbose --grep @pats",
```
Comment (dharrisMOJ) [inline package.json:25]: Should this be the tag?

File: src/test/end-to-end/config.js [hunk 2]
```diff
@@ -1,8 +1,8 @@
 module.exports = {
   TestUrl: process.env.TEST_E2E_URL || 'https://et-sya.aat.platform.hmcts.net',
-  TestShowBrowserWindow: process.env.SHOW_BROWSER_WINDOW || false,
+  TestShowBrowserWindow: process.env.SHOW_BROWSER_WINDOW || true,
```
Comment (dharrisMOJ) [inline src/test/end-to-end/config.js]: Set to false

File: src/test/end-to-end/config.js [hunk 3]
```diff
@@ -1,8 +1,8 @@
...
   TestsPathToRun: process.env.E2E_TEST_PATH || './features/**/*js',
   TestReportFolder: process.env.E2E_OUTPUT_DIR || './functional-output/reports',
-  TestEnvETUser: process.env.TEST_CASE_USERNAME || '',
-  TestEnvETPassword: process.env.TEST_CASE_PASSWORD || '',
+  TestEnvETUser: process.env.TEST_CASE_USERNAME || 'tester@hmcts.net1465',
```
Comment (dharrisMOJ) [inline src/test/end-to-end/config.js]: Remove user and password

File: src/test/end-to-end/features/EmploymentRespondentDetails/verifyDidYouWorkForOrg.js [hunk 4]
```diff
@@ -8,11 +8,11 @@ Scenario("Did you work for the organisation or person you're making your claim a
   await I.authenticateWithIdam();
   await doNotHaveToCompleteCard(I);
   await didYouWorkForOrganisation(I, 'No');
-}).tag('@RET-WIP');
+}).tag('@RET-BAT');
```
Comment (dharrisMOJ) [inline src/test/end-to-end/features/EmploymentRespondentDetails/verifyDidYouWorkForOrg.js:11]: Should this be the tag?

File: src/test/end-to-end/pages/claimDetails/claimDet.js [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+    I.see('What to write for discrimination claims');
+    I.see('What to write for dismissal claims');
+    I.see('What to write for whistleblower claims');
+    I.see("What to write for 'other' claims");
+    I.see('Use this box to describe the events around your dispute, or add to your claim by uploading a document');
+    I.see('You have 2500 characters remaining');
+    I.see('Or upload your summary as a separate document (optional)');
+
+    I.click(claimDetailsConfig.what_to_write_for_discrimination_claims);
+    I.wait(1);
```
Comment (dharrisMOJ) [inline src/test/end-to-end/pages/claimDetails/claimDet.js:59]: Is the test running slow? do we need to add waits?

File: src/test/end-to-end/features/claimDetails/claimDetailsTest.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
+const { doNotHaveToCompleteCard, stepsToMakingYourClaim, claimDetails } = require('../../helpers/caseHelper');
+const commonFlow = require('../../helpers/commonFlow.js');
+const { I } = inject();
+
+Feature('Enter claim details for a claim');
+
+Scenario('Navigate Claim Details - Describe What happenned to you', async () => {
+  await commonFlow.createSingleMyselfCase();
```
Comment (dharrisMOJ) [inline src/test/end-to-end/features/claimDetails/claimDetailsTest.js:8]: I was thinking for these tests we navigate to the page directly or to the task list page. The e2e steps will cover the e2e journey. https://github.com/hmcts/et-sya-frontend/blob/master/src/test/end-to-end/features/e2e_Scenarios/submitClaimForSinglePerson.js

File: src/test/end-to-end/pages/stepsToMakingYourClaim/stepsToMakingYourClaim.js [hunk 7]
```diff
@@ -0,0 +1,36 @@
+'use strict';
+const contactUs = require('../../helpers/contactUs.js');
+
+module.exports = async function () {
+  // eslint-disable-next-line @typescript-eslint/no-this-alias
```
Comment (dharrisMOJ) [inline src/test/end-to-end/pages/stepsToMakingYourClaim/stepsToMakingYourClaim.js:5]: Do we need this function?

File: src/test/end-to-end/features/claimDetails/claimDetailsTest.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  await doNotHaveToCompleteCard(I);
+  await stepsToMakingYourClaim(I);
+  I.click("//a[contains(.,'Describe what happened to you')]");
+  await claimDetails(I);
+}).tag('@pats');
+
+Scenario('Navigate Claim Details - Tell us what you want from your claim', async () => {
+  await commonFlow.createSingleMyselfCase();
+  await I.authenticateWithIdam();
+  await doNotHaveToCompleteCard(I);
```
Comment (dharrisMOJ) [inline src/test/end-to-end/features/claimDetails/claimDetailsTest.js:19]: See above, if navigating directly to the page we don't need this function

File: src/test/end-to-end/features/claimDetails/claimDetailsTest.js [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+  await stepsToMakingYourClaim(I);
+  I.click("//a[contains(.,'Tell us what you want from your claim')]");
+  await claimDetails(I, false);
```
Comment (dharrisMOJ) [inline src/test/end-to-end/features/claimDetails/claimDetailsTest.js:22]: Will this be the function used and added to our e2e test submitClaimForSinglePerson.js?

File: src/test/end-to-end/config.js (hunk 2 above)
Comment (pats-john) [reply src/test/end-to-end/config.js]: Addressed...

File: src/test/end-to-end/config.js (hunk 3 above)
Comment (pats-john) [reply src/test/end-to-end/config.js]: Removed...

File: src/test/end-to-end/features/EmploymentRespondentDetails/verifyDidYouWorkForOrg.js (hunk 4 above)
Comment (pats-john) [reply src/test/end-to-end/features/EmploymentRespondentDetails/verifyDidYouWorkForOrg.js:11]: This is not my test but I can revert the tag if it helps...

File: src/test/end-to-end/pages/claimDetails/claimDet.js (hunk 5 above)
Comment (pats-john) [reply src/test/end-to-end/pages/claimDetails/claimDet.js:59]: The Waits are put in for the links to be expanded before the tests proceed...I can try to comment this and run few tests  on them  and see...

File: package.json (hunk 1 above)
Comment (pats-john) [reply package.json]: Yes this is changed now...

**Review by AnishkumarMistry — APPROVED** [review]

File: src/test/end-to-end/features/claimDetails/claimDetailsTest.js (hunk 6 above)
Comment (pats-john) [reply src/test/end-to-end/features/claimDetails/claimDetailsTest.js]: This has been updated as requested...

File: src/test/end-to-end/pages/stepsToMakingYourClaim/stepsToMakingYourClaim.js (hunk 7 above)
Comment (pats-john) [reply src/test/end-to-end/pages/stepsToMakingYourClaim/stepsToMakingYourClaim.js:5]: yes this page would be tested as part of the claim Details and would be usefull for future end-2-end tests as well...

File: src/test/end-to-end/features/claimDetails/claimDetailsTest.js (hunk 8 above)
Comment (pats-john) [reply src/test/end-to-end/features/claimDetails/claimDetailsTest.js]: yes the test has been modified to this effect...

File: src/test/end-to-end/features/claimDetails/claimDetailsTest.js (hunk 9 above)
Comment (pats-john) [reply src/test/end-to-end/features/claimDetails/claimDetailsTest.js:23]: Yes ...This has been added ...

File: src/test/end-to-end/pages/claimDetails/claimDet.js (hunk 5 above)
Comment (pats-john) [reply src/test/end-to-end/pages/claimDetails/claimDet.js]: This has been implemented as well...

**Review by SunnyAyeni — APPROVED** [review]

**Pull request merged**